app/services/flow/flows/provider_means_state_benefits.rb

```diff
@@ -0,0 +1,45 @@
+module Flow
+  module Flows
+    class ProviderMeansStateBenefits < FlowSteps
+      STEPS = {
+        receives_state_benefits: {
+          path: ->(application) { urls.providers_legal_aid_application_means_receives_state_benefits_path(application) },
+          forward: lambda do |_application, receives_state_benefits|
+            receives_state_benefits ? :state_benefits : :regular_incomes
+          end,
+          check_answers: lambda do |application|
+            if application.applicant.receives_state_benefits?
+              if application.applicant.state_benefits.count.positive?
+                :add_other_state_benefits
+              else
+                :state_benefits
+              end
+            else
+              :check_income_answers
+            end
+          end,
+        },
+        state_benefits: {
+          path: ->(application) { urls.new_providers_legal_aid_application_means_state_benefit_path(application) },
+          forward: :add_other_state_benefits,
+        },
+        add_other_state_benefits: {
+          path: ->(application) { urls.providers_legal_aid_application_means_add_other_state_benefits_path(application) },
+          forward: lambda do |_application, add_other_state_benefits|
+            add_other_state_benefits ? :state_benefits : :regular_incomes
+          end,
+          check_answers: lambda do |_application, add_other_state_benefits|
+            add_other_state_benefits ? :state_benefits : :check_income_answers
+          end,
+        },
+        remove_state_benefits: {
+          forward: lambda do |application|
+            transaction_type = TransactionType.find_by(name: "benefits")
+            state_benefit_transactions = application.regular_transactions.where(transaction_type_id: transaction_type.id)
```

Comment:
Just a suggestion but maybe this query/logic could be put in the controller and just pass a variable 'has_other_benefits' or something in the forward method? It seems a bit complex for the flow imo and I don't think we have similar stuff elsewhere - up to you though :)

Reply:
I think this was an early addition to get it working... I can probably extract it from application.applicant.state_benefits now 🤔

Reply:
Does this seem better?
```ruby
       remove_state_benefits: {
          forward: lambda do |application|
            state_benefit_transactions = application.applicant.state_benefits
            state_benefit_transactions.any? ? :add_other_state_benefits : :receives_state_benefits
          end,
        },
```